### DISCUSSION TITLE
java.lang.Exception: BatchMakeP12 failed for 'tomcat'.

### DISCUSSION THREAD

**Discussion by pkiman2** [Keyfactor/ejbca-ce#703] [Q&A]
Hi,

Been pulling my hair with this new error I got as I am trying to build another ManagementCA from scratch. When I run ant runinstall, I get this : java.lang.Exception: BatchMakeP12 failed for 'tomcat'.

I am using the same commands I used when I created a successful build and deployment. I am completely puzzled with this new error. Attached are the logs and greatly appreciated for any help! :) 

[output-deployear.txt](https://github.com/user-attachments/files/17634123/output-deployear.txt)
[output-runtinstall.txt](https://github.com/user-attachments/files/17634124/output-runtinstall.txt)
[Wildfly-server.log](https://github.com/user-attachments/files/17634120/Wildfly-server.log)

**Comment by primetomas** [comment]
Two things.
1. Don't use sudo, there is no need to be root to build and deploy EJBCA. deploy and runinstall must be done as the OS user that WildFly runs as (or at least WildFly user must be able to read/write files written by the OS user used to deploy/install).
2. You can not so "runinstall" on an already installed system. Your system already have data in the database from a previous installation. Either you want to re-install in which case you have to delete the old database, or you simply create a new CA in EJBCA and transition to that CA, deleting the old Management CA when you are done.

**Reply by pkiman2** [reply]
Hi Tomas,

I was able to get past this error. I changed the hostname and the management CA name as they were the same. Then I dropped the initially created ejbca database and created a new one. But then raninstall threw another error: cannot contact EJBCA instance. Now, after reading your response, I ran through the file ownership for EJBCA_HOME and noticed the following:

1. the "dist" and "tmp" folder is owned by root: 

> xyz-admin@EJBCA01:~/ejbca/ejbca-ce$ ls -al
> total 324
> drwxrwxr-x 14 xyz-admin xyz-admin  4096 Nov  6 20:22 .
> drwxrwxr-x  3 xyz-admin xyz-admin  4096 Nov  5 15:54 ..
> drwxrwxr-x  4 xyz-admin xyz-admin  4096 Nov  5 15:55 bin
> -rw-rw-r--  1 xyz-admin xyz-admin 12856 Nov  5 15:55 build.gradle.kts
> -rw-rw-r--  1 xyz-admin xyz-admin 78780 Nov  5 15:55 build.xml
> -rw-rw-r--  1 xyz-admin xyz-admin 17858 Nov  5 15:55 .classpath
> drwxrwxr-x  4 xyz-admin xyz-admin  4096 Nov  6 13:10 conf
> -rw-rw-r--  1 xyz-admin xyz-admin  2741 Nov  5 15:55 CONTRIBUTING.md
> -rw-rw-r--  1 xyz-admin xyz-admin  5053 Nov  5 15:55 deprecated.xml
> drwxr-xr-x  5 root      root       4096 Nov  6 20:24 dist
> drwxrwxr-x  7 xyz-admin xyz-admin  4096 Nov  5 15:55 doc
> -rw-rw-r--  1 xyz-admin xyz-admin 11419 Nov  5 15:55 docs.xmli
> -rw-rw-r--  1 xyz-admin xyz-admin   331 Nov  5 15:55 echo.xml
> drwxrwxr-x  8 xyz-admin xyz-admin  4096 Nov  5 15:55 .git
> drwxrwxr-x  4 xyz-admin xyz-admin  4096 Nov  5 15:55 .github
> -rw-rw-r--  1 xyz-admin xyz-admin   250 Nov  5 15:55 .gitignore
> drwxrwxr-x  3 xyz-admin xyz-admin  4096 Nov  5 15:55 gradle
> -rw-rw-r--  1 xyz-admin xyz-admin   207 Nov  5 15:55 gradle.properties
> -rwxrwxr-x  1 xyz-admin xyz-admin  8692 Nov  5 15:55 gradlew
> -rw-rw-r--  1 xyz-admin xyz-admin  2868 Nov  5 15:55 gradlew.bat
> drwxrwxr-x 10 xyz-admin xyz-admin  4096 Nov  5 18:44 lib
> -rw-rw-r--  1 xyz-admin xyz-admin 26436 Nov  5 15:55 LICENSE
> drwxrwxr-x 37 xyz-admin xyz-admin  4096 Nov  6 20:22 modules
> -rw-rw-r--  1 xyz-admin xyz-admin  1297 Nov  5 15:55 .project
> -rw-rw-r--  1 xyz-admin xyz-admin 11436 Nov  5 15:55 propertyDefaults.xml
> -rw-rw-r--  1 xyz-admin xyz-admin  4209 Nov  5 15:55 README.md
> -rw-rw-r--  1 xyz-admin xyz-admin  3776 Nov  5 15:55 removed.xml
> -rw-rw-r--  1 xyz-admin xyz-admin   778 Nov  5 15:55 SECURITY.md
> drwxrwxr-x  2 xyz-admin xyz-admin  4096 Nov  5 15:55 .settings
> -rw-rw-r--  1 xyz-admin xyz-admin  8160 Nov  5 15:55 settings.gradle.kts
> drwxrwxr-x 10 xyz-admin xyz-admin  4096 Nov  5 15:55 src
> -rw-rw-r--  1 xyz-admin xyz-admin 28836 Nov  5 15:55 test.xmli
> drwxr-xr-x  6 root      root       4096 Nov  6 20:24 tmp

2. Running ant -q clean deployear gives me this error:

> xyz-admin@EJBCA01:~/ejbca/ejbca-ce$ ant -q clean deployear
>      [echo] No custom changes to merge.
> 
> BUILD FAILED
> /home/xyz-admin/ejbca/ejbca-ce/build.xml:36: Unable to delete file /home/xyz-admin/ejbca/ejbca-ce/dist/clientToolBox/properties/internal.properties
> 
> Total time: 0 seconds

3. I changed the ownership to xyz-admin but still getting permission error, I guess:

> xyz-admin@EJBCA01:~/ejbca/ejbca-ce$ ant -q clean deployear
>      [echo] No custom changes to merge.
>      [echo] Cleaning cesecore-x509ca
>      [echo] Cleaning va
>      [echo] Cleaning ra
>      [echo] appserver.home: /opt/wildfly
>      [echo] appserver.type: jboss
>      [echo] appserver.subtype: jboss7
>      [echo] 
>      [echo] ---------- EJBCA 8.3.2 Enterprise (working copy) CONFIGURATION PROPERTIES ----------
>      [echo] appserver.type           = jboss
>      [echo] appserver.subtype        = jboss7
>      [echo] appserver.home           = /opt/wildfly
>      [echo] java.version(ant.java)   = 11.0.23 (11)
>      [echo] ocsp.signaturealgorithm  = SHA256WithRSA;SHA256withRSAandMGF1;SHA384WithRSA;SHA512WithRSA;SHA224withECDSA;SHA256withECDSA;SHA384withECDSA;SHA512withECDSA;Ed25519;Ed448
>      [echo] datasource.jndi-name     = EjbcaDS
>      [echo] datasource.jndi-name-prefix = java:/
>      [echo] database.name            = mysql
>      [echo] database.url             = jdbc:mysql://localhost:3306/ejbca?characterEncoding=UTF-8
>      [echo] database.driver          = org.mariadb.jdbc.Driver
>      [echo] database.username        = ejbca
>      [echo] database.password        = ***
>      [echo] database.useSeparateCertificateTable = false
>      [echo] database.valid.connection.sql = select 1
>      [echo] mail.jndi-name           = java:/EjbcaMail
>      [echo] mail.from                = ejbca-donotreply@domain.com
>      [echo] mail.user                = ejbca_user
>      [echo] mail.password            = ***
>      [echo] mail.smtp.host           = localhost
>      [echo] mail.smtp.port           = 25
>      [echo] mail.smtp.auth           = false
>      [echo] mail.debug               = false
>      [echo] httpserver.pubhttp       = 8080
>      [echo] httpserver.pubhttps      = 8442
>      [echo] httpserver.privhttps     = 8443
>      [echo] httpserver.external.privhttps = 8443
>      [echo] httpsserver.hostname     = EJBCA01
>      [echo] httpsserver.password     = ***
>      [echo] httpserver.external.fqdn =
>      [echo] web.contentencoding      = UTF-8
>      [echo] ejbcaws.enabled          = true
>      [echo] unidfnr.enabled          = false
>      [echo]
>      [echo]
>     [javac] Note: Some input files use or override a deprecated API.
>     [javac] Note: Recompile with -Xlint:deprecation for details.
>     [javac] Note: Some input files use unchecked or unsafe operations.
>     [javac] Note: Recompile with -Xlint:unchecked for details.
>     [javac] Note: /home/xyz-admin/ejbca/ejbca-ce/modules/cesecore-entity/src/org/cesecore/roles/RoleInformation.java uses or overrides a deprecated API.
>     [javac] Note: Recompile with -Xlint:deprecation for details.
>     [javac] Note: /home/xyz-admin/ejbca/ejbca-ce/modules/cesecore-ejb/src/org/cesecore/authorization/AuthorizationSessionBean.java uses or overrides a deprecated API.
>     [javac] Note: Recompile with -Xlint:deprecation for details.
>     [javac] Note: /home/xyz-admin/ejbca/ejbca-ce/modules/cesecore-ejb/src/org/cesecore/certificates/ca/CaSessionBean.java uses unchecked or unsafe operations.
>     [javac] Note: Recompile with -Xlint:unchecked for details.
>     [javac] Note: Some input files use or override a deprecated API.
>     [javac] Note: Recompile with -Xlint:deprecation for details.
>     [javac] Note: Some input files use or override a deprecated API.
>     [javac] Note: Recompile with -Xlint:deprecation for details.
>     [javac] Note: Some input files use or override a deprecated API.
>     [javac] Note: Recompile with -Xlint:deprecation for details.
>     [javac] Note: /home/xyz-admin/ejbca/ejbca-ce/modules/healthcheck-war/src/org/ejbca/ui/web/pub/VaPeerStatusServlet.java uses unchecked or unsafe operations.
>     [javac] Note: Recompile with -Xlint:unchecked for details.
>     [javac] Note: Some input files use or override a deprecated API.
>     [javac] Note: Recompile with -Xlint:deprecation for details.
>     [javac] Note: Some input files use or override a deprecated API.
>     [javac] Note: Recompile with -Xlint:deprecation for details.
>     [javac] Note: Some input files use or override a deprecated API.
>     [javac] Note: Recompile with -Xlint:deprecation for details.
>     [javac] Note: Some input files use or override a deprecated API.
>     [javac] Note: Recompile with -Xlint:deprecation for details.
>     [javac] Note: /home/xyz-admin/ejbca/ejbca-ce/modules/ra-gui/src/org/ejbca/ra/jsfext/RaExceptionHandlerFactory.java uses or overrides a deprecated API.
>     [javac] Note: Recompile with -Xlint:deprecation for details.
>     [javac] warning: Implicitly compiled files were not subject to annotation processing.
>     [javac]   Use -proc:none to disable annotation processing or -implicit to specify a policy for implicit compilation.
>     [javac] Note: /home/xyz-admin/ejbca/ejbca-ce/modules/clientToolBox/src/org/ejbca/ui/cli/SCEPTest.java uses or overrides a deprecated API.
>     [javac] Note: Recompile with -Xlint:deprecation for details.
>     [javac] 1 warning
>     [javac] warning: Implicitly compiled files were not subject to annotation processing.
>     [javac]   Use -proc:none to disable annotation processing or -implicit to specify a policy for implicit compilation.
>     [javac] 1 warning
>       [jar] JARs are never empty, they contain at least a manifest file
>      [echo] Using local documentation, if any. Set -Ddoc.update=true if you wish to update documentation from Confluence.
>      [echo] WS=/home/xyz-admin/ejbca/ejbca-ce/modules/ejbca-ws/src/org/ejbca/core/protocol/ws/objects
>      [echo] Folder found: true
>      [echo] Local documentation is now available in file:///home/xyz-admin/ejbca/ejbca-ce/tmp/htdocs/docs/index.html
>      [echo] in-test-mode: false
>      [echo] Enabled module doc.war
>      [echo] Enabled module ejbca-ws-ejb.jar
>      [echo] Disabled module systemtests-ejb.jar
>      [echo] Disabled module statedump-ejb.jar
>      [echo] Disabled module configdump-ejb.jar
>      [echo] Disabled module peerconnector-ejb.jar
>      [echo] Disabled module cesecore-cvcca.jar
>      [echo] Enabled module cesecore-x509ca.jar
>      [echo] Disabled module proxy-ca.jar
>      [echo] Disabled module unidfnr-ejb.jar
>      [echo] Disabled module peerconnector.rar
>      [echo] Disabled module peerconnector.war
>      [echo] Enabled module ra-gui.war
>      [echo] Disabled module acme.war
>      [echo] Disabled module msae.war
>      [echo] Disabled module cits.war
>      [echo] Disabled module est.war
>      [echo] Enabled module ejbca-rest-api.war
>      [echo] Disabled module swagger-ui.war
>      [echo] Disabled module ssh.war
>      [echo] Enabled module status.war
>      [echo] Enabled module certstore.war
>      [echo] Enabled module crlstore.war
>      [echo] Specify -Dsignjar.keystore=/path/keystore.jks if you want to sign the release.
>      [echo] Using appserver.home : /opt/wildfly
> 
> BUILD FAILED
> /home/xyz-admin/ejbca/ejbca-ce/build.xml:834: The following error occurred while executing this line:
> /home/xyz-admin/ejbca/ejbca-ce/bin/jboss.xml:99: Failed to copy /home/xyz-admin/ejbca/ejbca-ce/dist/ejbca.ear to /opt/wildfly/standalone/deployments/ejbca.ear due to org.apache.tools.ant.util.ResourceUtils$ReadOnlyTargetFileException can't write to read-only destination file /opt/wildfly/standalone/deployments/ejbca.ear
> 
> Total time: 39 seconds

4. I went in to :/opt/wildfly/standalone/deployments and removed the previously built ear files:

> xyz-admin@EJBCA01:~/ejbca/ejbca-ce$ cd /opt/wildfly/standalone/deployments
> xyz-admin@EJBCA01:/opt/wildfly/standalone/deployments$ ls
> ejbca.ear  ejbca.ear.deployed  mariadb-java-client.jar  mariadb-java-client.jar.deployed  README.txt
> xyz-admin@EJBCA01:/opt/wildfly/standalone/deployments$ rm ejbca.ear
> rm: remove write-protected regular file 'ejbca.ear'? y
> rm: cannot remove 'ejbca.ear': Permission denied
> xyz-admin@EJBCA01:/opt/wildfly/standalone/deployments$ sudo rm ejbca.ear
> [sudo] password for xyz-admin: 
> xyz-admin@EJBCA01:/opt/wildfly/standalone/deployments$ 
> xyz-admin@EJBCA01:/opt/wildfly/standalone/deployments$ ls -al
> total 640
> drwxr-xr-x 2 wildfly wildfly   4096 Nov  7 13:51 .
> drwxr-xr-x 8 wildfly wildfly   4096 Nov  5 18:38 ..
> -rw-r--r-- 1 wildfly wildfly      9 Nov  6 20:24 ejbca.ear.deployed
> -rw-r--r-- 1 wildfly wildfly 624101 Sep 25  2020 mariadb-java-client.jar
> -rw-r--r-- 1 wildfly wildfly     23 Sep 25  2020 mariadb-java-client.jar.deployed
> -rw-r--r-- 1 wildfly wildfly   8888 Dec 14  2021 README.txt
> xyz-admin@EJBCA01:/opt/wildfly/standalone/deployments$ sudo rm ejbca.ear.deployed 
> xyz-admin@EJBCA01:/opt/wildfly/standalone/deployments$ ls -al
> total 636
> drwxr-xr-x 2 wildfly wildfly   4096 Nov  7 13:51 .
> drwxr-xr-x 8 wildfly wildfly   4096 Nov  5 18:38 ..
> -rw-r--r-- 1 wildfly wildfly 624101 Sep 25  2020 mariadb-java-client.jar
> -rw-r--r-- 1 wildfly wildfly     23 Sep 25  2020 mariadb-java-client.jar.deployed
> -rw-r--r-- 1 wildfly wildfly   8888 Dec 14  2021 README.txt

5. Ran deployear again but still getting the same results:

> xyz-admin@EJBCA01:/opt/wildfly/standalone/deployments$ cd ${EJBCA_HOME}
> xyz-admin@EJBCA01:~/ejbca/ejbca-ce$ ant -q clean deployear
>      [echo] No custom changes to merge.
>      [echo] Cleaning cesecore-x509ca
>      [echo] Cleaning va
>      [echo] Cleaning ra
>      [echo] appserver.home: /opt/wildfly
>      [echo] appserver.type: jboss
>      [echo] appserver.subtype: jboss7
>      [echo] 
>      [echo] ---------- EJBCA 8.3.2 Enterprise (working copy) CONFIGURATION PROPERTIES ----------
>      [echo] appserver.type           = jboss
>      [echo] appserver.subtype        = jboss7
>      [echo] appserver.home           = /opt/wildfly
>      [echo] java.version(ant.java)   = 11.0.23 (11)
>      [echo] ocsp.signaturealgorithm  = SHA256WithRSA;SHA256withRSAandMGF1;SHA384WithRSA;SHA512WithRSA;SHA224withECDSA;SHA256withECDSA;SHA384withECDSA;SHA512withECDSA;Ed25519;Ed448
>      [echo] datasource.jndi-name     = EjbcaDS
>      [echo] datasource.jndi-name-prefix = java:/
>      [echo] database.name            = mysql
>      [echo] database.url             = jdbc:mysql://localhost:3306/ejbca?characterEncoding=UTF-8
>      [echo] database.driver          = org.mariadb.jdbc.Driver
>      [echo] database.username        = ejbca
>      [echo] database.password        = ***
>      [echo] database.useSeparateCertificateTable = false
>      [echo] database.valid.connection.sql = select 1
>      [echo] mail.jndi-name           = java:/EjbcaMail
>      [echo] mail.from                = ejbca-donotreply@domain.com
>      [echo] mail.user                = ejbca_user
>      [echo] mail.password            = ***
>      [echo] mail.smtp.host           = localhost
>      [echo] mail.smtp.port           = 25
>      [echo] mail.smtp.auth           = false
>      [echo] mail.debug               = false
>      [echo] httpserver.pubhttp       = 8080
>      [echo] httpserver.pubhttps      = 8442
>      [echo] httpserver.privhttps     = 8443
>      [echo] httpserver.external.privhttps = 8443
>      [echo] httpsserver.hostname     = EJBCA01
>      [echo] httpsserver.password     = ***
>      [echo] httpserver.external.fqdn =
>      [echo] web.contentencoding      = UTF-8
>      [echo] ejbcaws.enabled          = true
>      [echo] unidfnr.enabled          = false
>      [echo]
>      [echo]
>     [javac] Note: Some input files use or override a deprecated API.
>     [javac] Note: Recompile with -Xlint:deprecation for details.
>     [javac] Note: Some input files use unchecked or unsafe operations.
>     [javac] Note: Recompile with -Xlint:unchecked for details.
>     [javac] Note: /home/xyz-admin/ejbca/ejbca-ce/modules/cesecore-entity/src/org/cesecore/roles/RoleInformation.java uses or overrides a deprecated API.
>     [javac] Note: Recompile with -Xlint:deprecation for details.
>     [javac] Note: /home/xyz-admin/ejbca/ejbca-ce/modules/cesecore-ejb/src/org/cesecore/authorization/AuthorizationSessionBean.java uses or overrides a deprecated API.
>     [javac] Note: Recompile with -Xlint:deprecation for details.
>     [javac] Note: /home/xyz-admin/ejbca/ejbca-ce/modules/cesecore-ejb/src/org/cesecore/certificates/ca/CaSessionBean.java uses unchecked or unsafe operations.
>     [javac] Note: Recompile with -Xlint:unchecked for details.
>     [javac] Note: Some input files use or override a deprecated API.
>     [javac] Note: Recompile with -Xlint:deprecation for details.
>     [javac] Note: Some input files use or override a deprecated API.
>     [javac] Note: Recompile with -Xlint:deprecation for details.
>     [javac] Note: Some input files use or override a deprecated API.
>     [javac] Note: Recompile with -Xlint:deprecation for details.
>     [javac] Note: /home/xyz-admin/ejbca/ejbca-ce/modules/healthcheck-war/src/org/ejbca/ui/web/pub/VaPeerStatusServlet.java uses unchecked or unsafe operations.
>     [javac] Note: Recompile with -Xlint:unchecked for details.
>     [javac] Note: Some input files use or override a deprecated API.
>     [javac] Note: Recompile with -Xlint:deprecation for details.
>     [javac] Note: Some input files use or override a deprecated API.
>     [javac] Note: Recompile with -Xlint:deprecation for details.
>     [javac] Note: Some input files use or override a deprecated API.
>     [javac] Note: Recompile with -Xlint:deprecation for details.
>     [javac] Note: Some input files use or override a deprecated API.
>     [javac] Note: Recompile with -Xlint:deprecation for details.
>     [javac] Note: /home/xyz-admin/ejbca/ejbca-ce/modules/ra-gui/src/org/ejbca/ra/jsfext/RaExceptionHandlerFactory.java uses or overrides a deprecated API.
>     [javac] Note: Recompile with -Xlint:deprecation for details.
>     [javac] warning: Implicitly compiled files were not subject to annotation processing.
>     [javac]   Use -proc:none to disable annotation processing or -implicit to specify a policy for implicit compilation.
>     [javac] Note: /home/xyz-admin/ejbca/ejbca-ce/modules/clientToolBox/src/org/ejbca/ui/cli/SCEPTest.java uses or overrides a deprecated API.
>     [javac] Note: Recompile with -Xlint:deprecation for details.
>     [javac] 1 warning
>     [javac] warning: Implicitly compiled files were not subject to annotation processing.
>     [javac]   Use -proc:none to disable annotation processing or -implicit to specify a policy for implicit compilation.
>     [javac] 1 warning
>       [jar] JARs are never empty, they contain at least a manifest file
>      [echo] Using local documentation, if any. Set -Ddoc.update=true if you wish to update documentation from Confluence.
>      [echo] WS=/home/xyz-admin/ejbca/ejbca-ce/modules/ejbca-ws/src/org/ejbca/core/protocol/ws/objects
>      [echo] Folder found: true
>      [echo] Local documentation is now available in file:///home/xyz-admin/ejbca/ejbca-ce/tmp/htdocs/docs/index.html
>      [echo] in-test-mode: false
>      [echo] Enabled module doc.war
>      [echo] Enabled module ejbca-ws-ejb.jar
>      [echo] Disabled module systemtests-ejb.jar
>      [echo] Disabled module statedump-ejb.jar
>      [echo] Disabled module configdump-ejb.jar
>      [echo] Disabled module peerconnector-ejb.jar
>      [echo] Disabled module cesecore-cvcca.jar
>      [echo] Enabled module cesecore-x509ca.jar
>      [echo] Disabled module proxy-ca.jar
>      [echo] Disabled module unidfnr-ejb.jar
>      [echo] Disabled module peerconnector.rar
>      [echo] Disabled module peerconnector.war
>      [echo] Enabled module ra-gui.war
>      [echo] Disabled module acme.war
>      [echo] Disabled module msae.war
>      [echo] Disabled module cits.war
>      [echo] Disabled module est.war
>      [echo] Enabled module ejbca-rest-api.war
>      [echo] Disabled module swagger-ui.war
>      [echo] Disabled module ssh.war
>      [echo] Enabled module status.war
>      [echo] Enabled module certstore.war
>      [echo] Enabled module crlstore.war
>      [echo] Specify -Dsignjar.keystore=/path/keystore.jks if you want to sign the release.
>      [echo] Using appserver.home : /opt/wildfly
> Attempt to copy /home/xyz-admin/ejbca/ejbca-ce/dist/ejbca.ear to /opt/wildfly/standalone/deployments/ejbca.ear using NIO Channels failed due to '/opt/wildfly/standalone/deployments/ejbca.ear'.  Falling back to streams.
> 
> BUILD FAILED
> /home/xyz-admin/ejbca/ejbca-ce/build.xml:834: The following error occurred while executing this line:
> /home/xyz-admin/ejbca/ejbca-ce/bin/jboss.xml:99: Failed to copy /home/xyz-admin/ejbca/ejbca-ce/dist/ejbca.ear to /opt/wildfly/standalone/deployments/ejbca.ear due to java.nio.file.AccessDeniedException /opt/wildfly/standalone/deployments/ejbca.ear
> 
> Total time: 36 seconds

I guess 99% of my errors are due to file permissions but I am entirely clueless as to why dist and tmp folders were owned by root. I was never root or use sudo cloning the repo and as you previously taught me, changed the /opt/wildfly ownership to wildfly. From what the error is giving me, it looks like xyz-admin cannot write the ear file to deployment folder because of permission. 

Should I execute the commands as wildfly user instead of xyz-admin? Not sure if this is possible as wildfly user uses its own hashed password as defined in the documentation (elytron store). Any advice is highly appreciated. Thank you!

**Reply by primetomas** [reply]
/opt/wildfly is probably a link, make sure you change permissions of ALL files in the directory tree, and not only on the symlink. Everything under /opt/wildfly should be owned by the wildfly user. 
deployear and runinstall should be run as the wildfly user.

**Reply by pkiman2** [reply]
Yup. Everything in /opt/wildfly and /opt/wildfly-26.0.0.Final are all owned by wildfly user:

```
xyz-admin@EJBCA01:~/ejbca/ejbca-ce$ ls -al /opt/wildfly-26.0.0.Final
total 544
drwxr-xr-x 11 wildfly wildfly   4096 Dec 14  2021 .
drwxr-xr-x  3 root    root      4096 Nov  5 18:34 ..
drwxr-xr-x  3 wildfly wildfly   4096 Dec 14  2021 appclient
drwxr-xr-x  3 wildfly wildfly   4096 Nov  5 18:37 bin
-rw-r--r--  1 wildfly wildfly  19358 Dec 14  2021 copyright.txt
drwxr-xr-x  6 wildfly wildfly   4096 Dec 14  2021 docs
drwxr-xr-x  4 wildfly wildfly   4096 Dec 14  2021 domain
drwxr-xr-x  2 wildfly wildfly   4096 Dec 14  2021 .installation
-rw-r--r--  1 wildfly wildfly 456439 Dec 14  2021 jboss-modules.jar
-rw-r--r--  1 wildfly wildfly  26530 Dec 14  2021 LICENSE.txt
drwxr-xr-x  3 wildfly wildfly   4096 Dec 14  2021 modules
-rw-r--r--  1 wildfly wildfly   2162 Dec 14  2021 README.txt
drwxr-xr-x  8 wildfly wildfly   4096 Nov  5 18:38 standalone
drwxr-xr-x  2 wildfly wildfly   4096 Dec 14  2021 welcome-content
drwxr-xr-x  3 wildfly wildfly   4096 Dec 14  2021 .well-known
xyz-admin@EJBCA01:~/ejbca/ejbca-ce$ ls -al /opt/wildfly
lrwxrwxrwx 1 wildfly wildfly 25 Nov  5 18:34 /opt/wildfly -> /opt/wildfly-26.0.0.Final
```

When you say "deployear and runinstall should be run as the wildfly user", I tried logging in as wildfly and I used the one in /usr/bin/wildfly_pass and it didn't work. Could be hashed. I am deeply sorry for being so dumb on this. But how can I get the wildfly password?

**Reply by primetomas** [reply]
running as wildfly user means "sudo su - wildfly" or similar.

Run:
ls -al /opt/wildfly/standalone/deployments/

**Reply by pkiman2** [reply]
xyz-admin@EJBCA01:~/ejbca/ejbca-ce$ ls -al /opt/wildfly/standalone/deployments/
total 636
drwxr-xr-x 2 wildfly wildfly   4096 Nov  7 13:51 .
drwxr-xr-x 8 wildfly wildfly   4096 Nov  5 18:38 ..
-rw-r--r-- 1 wildfly wildfly 624101 Sep 25  2020 mariadb-java-client.jar
-rw-r--r-- 1 wildfly wildfly     23 Sep 25  2020 mariadb-java-client.jar.deployed
-rw-r--r-- 1 wildfly wildfly   8888 Dec 14  2021 README.txt
